cdk/kraken/src/deploy.ts

```diff
@@ -37,50 +43,98 @@ export class DeployJob extends CheckoutJob {
     const fullConfig: Required<DeployJobProps> = {
       deployTag: "${{ github.sha }}",
       defaultBranch: "master",
+      deployToFeatureBranch: false,
       ...config,
     };
 
-    super(scope, "deploy", {
-      runsOn: "ubuntu-latest",
-      if: `github.ref == 'refs/heads/${fullConfig.defaultBranch}'`,
-      steps: [
-        {
-          id: "synth",
-          name: "Synth cdk8s manifests",
-          run: dedent`cd k8s
+    const featureDeploySynthSteps: StepsProps[] = [
+      {
+        name: "Get Pull Request Number",
+        id: "pr",
+        run: dedent`
+          echo "::set-output name=pull_request_number::$(gh pr view --json number -q .number || echo "")"
+                  `,
+        env: {
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}",
+        },
+      },
+      {
+        id: "synth",
+        name: "Synth cdk8s manifests",
+        if: "steps.pr.outputs.pull_request_number",
+        run: dedent`cd k8s
           yarn install --frozen-lockfile
+          
+          # Get repo name (by removing owner/organization)
+          export DEPLOY_TO_FEATURE_BRANCH=true
+          export RELEASE_NAME=${"${REPOSITORY#*/}-pr-${{ steps.pr.outputs.pull_request_number }}"}
```

Comment:
You may be wondering, "Joy, why are you doing this dumb thing `RELEASE_NAME=${"${REPOSITORY#*/}}`? Couldn't you just escape it using `RELEASE_NAME=\${REPOSITORY#*/}`

The answer is that `ts-dedent` is clowning me. I spent too much time trying to figure out what is going on. This is a bug, so I'm using this very sad format to do it. Hopefully someday they fix it and I'll use the clean way.